libraries/AP_Baro/AP_Baro.cpp

```diff
@@ -543,8 +623,31 @@ void AP_Baro::update(void)
             }
             // sanity check altitude
             sensors[i].alt_ok = !(isnan(altitude) || isinf(altitude));
+
             if (sensors[i].alt_ok) {
-                sensors[i].altitude = altitude + _alt_offset_active;
+                altitude += _alt_offset_active;
+
+                if (process_gps_alt) {
+                    // calculation of actual error based on current and calibrated GPS altitude
+                    const float gps_err = gps_alt - _gps_calibration_altitude - altitude;
+
+                    // limitation of error magnitude
+                    float prev_err = _gps_alt_error[i].get();
+                    const float constr_gps_err_diff = constrain_value(gps_err - prev_err, -BARO_GPS_MAX_ERR, BARO_GPS_MAX_ERR);
```

Comment:
First one is actually meant to be spikes prevention and works with predefined constrain. It does limit the value that is fed into IIR.
Second limit is set by user and limits the rate of change of the output of IIR.

From your post on previous PR:

> The calculation of the baro drift correction should perform the following processing order:
a) Calculate the error between the reference and baro alt
b) Clamped the magnitude of the error to prevent large spikes driving downstream IIR filters
c) Apply an IIR LPF to the error using ADJ_TC_DEFAULT
d) Rate limit the filtered error using ADJ_RATE_DEFAULT to calculate the correction

So the first constrain is doing b and second one is doing d.